src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Metadata/SourceGenJsonTypeInfoOfT.cs

```diff
@@ -110,11 +110,11 @@ internal override JsonParameterInfoValues[] GetParameterInfoValues()
             return array;
         }
 
-        internal void AddPropertiesUsingSourceGenInfo()
+        internal JsonPropertyDictionary<JsonPropertyInfo>? AddPropertiesUsingSourceGenInfo()
         {
             if (PropertyInfoForTypeInfo.ConverterStrategy != ConverterStrategy.Object)
             {
-                return;
+                return null;
```

Comment:
Can we also exit early for `typeof(T)`?

Reply:
likely yes but I'd prefer to keep those changes separate